timewarp_gl/plugin.cpp

```diff
@@ -620,10 +662,43 @@ class timewarp_gl : public threadloop {
 
 	virtual ~timewarp_gl() override {
 		// TODO: Need to cleanup resources here!
-		glXMakeCurrent(xwin->dpy, None, NULL);
+		assert(errno == 0 && "Errno should not be set at start of destructor");
+
+		if (!glXMakeCurrent(xwin->dpy, None, nullptr)) {
+			std::cerr << "glXMakeCurrent returned false in " << __FILE__ << ":" << __LINE__ << std::endl;
+            std::exit(1);
+		}
+
+        RAC_ERRNO_MSG("timewarp_gl before closing X");
  		glXDestroyContext(xwin->dpy, xwin->glc);
  		XDestroyWindow(xwin->dpy, xwin->win);
  		XCloseDisplay(xwin->dpy);
+        RAC_ERRNO_MSG("timewarp_gl after closing X");
+
+	    if (distortion_indices != nullptr) {
```

Comment:
`free` is good, but `std::vector` is better for these variables.

I think we want them to be a vector of vectors, one for each eye. This eliminates `vec[eye*NUM_ELEM + elem]`.